test/e2e/lib/driver-helper.js

```diff
@@ -611,3 +613,89 @@ export async function waitTillTextPresent( driver, selector, text, waitOverride
 		`Timed out waiting for element with ${ selector.using } of '${ selector.value }' to be present and displayed with text '${ text }'`
 	);
 }
+
+async function fetchTranslations( originals, locale, project = 'wpcom' ) {
+	return request
+		.post( 'https://translate.wordpress.com/api/translations/-query-by-originals', {
+			form: {
+				project,
+				locale_slug: locale,
+				original_strings: JSON.stringify( originals ),
+			},
+		} )
+		.then( ( response ) => {
+			let translations = JSON.parse( response );
+			delete translations.originals_not_found;
+			translations = Object.values( translations );
+
+			return translations;
+		} )
+		.catch( () => null );
+}
+
+async function getElementsTranslations( elements, locale, project ) {
+	// Default locale doesn't have translations
+	if ( locale === 'en' ) {
+		return null;
+	}
+
+	const originals = await webdriver.promise.map( elements, async ( element ) => {
+		const singular = await element.getAttribute( 'data-e2e-string' );
+		return { singular };
+	} );
+
+	return fetchTranslations( originals, locale, project );
+}
+
+export async function verifyTranslationsPresent(
+	driver,
+	locale,
+	selector = '[data-e2e-string]',
+	waitOverride
+) {
+	const timeoutWait = waitOverride ? waitOverride : explicitWaitMS;
+	const translatableElements = await webdriver.promise.filter(
+		await driver.findElements( by.css( selector ) ),
+		Boolean
+	);
+	const translations = await getElementsTranslations( translatableElements, locale );
+
+	for ( const element of translatableElements ) {
+		const singular = await element.getAttribute( 'data-e2e-string' );
+		const params = await element.getAttribute( 'data-e2e-string-params' );
+		let translation = singular;
+
+		// Translation for default locale should match the original
+		if ( locale !== 'en' ) {
+			const translationEntry =
+				translations && translations.find( ( entry ) => entry.original.singular === singular );
+			translation =
+				translationEntry &&
+				translationEntry.translations &&
+				translationEntry.translations[ 0 ] &&
+				translationEntry.translations[ 0 ].translation_0;
+		}
+
+		if ( params ) {
+			translation = sprintf( translation, ...JSON.parse( params ) );
+		}
+
+		await driver.wait(
+			function () {
+				return ( async () => {
+					// Non-breaking space characters are being replaced with regular space characters,
+					// due to an inconsistency in `element.getText()` causing it to return the text with
+					// the non-breaking space characters converted in some cases.
+					const elementTextSanitized = ( await element.getText() )
+						.trim()
+						.replace( /[\u00A0]/g, ' ' );
+					const translationSanitized = translation.trim().replace( /[\u00A0]/g, ' ' );
+
+					return elementTextSanitized === translationSanitized;
+				} )();
+			},
+			timeoutWait,
+			`Timed out waiting for element with translatable string '${ singular }' to be displayed with translation '${ translation }'`
```

Comment:
Do we need this to be wrapped with a `wait` loop? This would mean that the [callback func](https://github.com/Automattic/wp-calypso/pull/51000/files#diff-4b1fd2b061afbd8739cfc7b385f436817a04cb31125fadcda287711048263968R685-R694) is going to be repeatedly called until returning `true`/timeout error. Is there a case that `elementTextSanitized === translationSanitized` would initially return false?

Reply:
With asynchronously loaded translations, in theory it's possible that the strings may initially render in English.